jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecs/exec/EcsExecUtils.kt

```diff
@@ -24,12 +26,20 @@ import software.aws.toolkits.telemetry.Result
 object EcsExecUtils : CoroutineScope by ApplicationThreadPoolScope("EcsExec") {
     fun updateExecuteCommandFlag(project: Project, service: Service, enabled: Boolean) {
         launch {
-            val request = UpdateServiceRequest.builder()
-                .cluster(service.clusterArn())
-                .service(service.serviceName())
-                .enableExecuteCommand(enabled)
-                .forceNewDeployment(true).build()
-            project.awsClient<EcsClient>().updateService(request)
+            try {
+                val request = UpdateServiceRequest.builder()
+                    .cluster(service.clusterArn())
+                    .service(service.serviceName())
+                    .enableExecuteCommand(enabled)
+                    .forceNewDeployment(true).build()
+                project.awsClient<EcsClient>().updateService(request)
+                checkServiceState(project, service, enable = enabled)
+            } catch (e: InvalidParameterException) {
+                runInEdt {
+                    TaskRoleNotFoundWarningDialog(project).show()
+                    EcsTelemetry.enableExecuteCommand(project, Result.Failed)
```

Comment:
where do we have the Success published?

Reply:
After the call has been made and the service has reached final state